File: Chronos.App/ClientApp/src/app/models/rules/feature-rules.ts
import { StoryRules } from "./story-rules";
import { TaskRules } from "./task-rules";

export class FeatureRules {
    id = 0;
    name = '';
    zeroStoryRules: StoryRules = null;
    zeroTaskRules: TaskRules = null;
    featureTaskRules: TaskRules = null;
    storyRules: StoryRules = null;
    taskRules: TaskRules = null;

    constructor(obj?: Partial<FeatureRules>) {
        if (obj)
            Object.assign(this, obj);
    }
}

File: Chronos.App/DataContracts/FeatureRules/FeatureRules.cs
namespace Chronos.App.DataContracts.FeatureRules
{
    public class FeatureRules
    {
        public string Name { get; set; }
        public TaskRules FeatureTaskRules { get; set; }
        public StoryRules ZeroStoryRules { get; set; }
        public TaskRules ZeroTaskRules { get; set; }
        public StoryRules StoryRules { get; set; }
        public TaskRules TaskRules { get; set; }
    }
}
